src/main/java/org/opentripplanner/middleware/connecteddataplatform/AnonymizedTripRequest.java

```diff
@@ -137,55 +134,98 @@ private List<AnonymizedItinerary> getItineraries(
+                        anonymizedLeg.realTime = leg.realTime;
+                        anonymizedLeg.routeId = leg.routeId;
+                        anonymizedLeg.routeShortName = leg.routeShortName;
+                        anonymizedLeg.routeLongName = leg.routeLongName;
+                        anonymizedLeg.routeType = leg.routeType;
+                        anonymizedLeg.tripBlockId = leg.tripBlockId;
+                        anonymizedLeg.tripId = leg.tripId;
+                    } else {
+                        // Parameters for non transit leg.
+                        anonymizedLeg.rentedVehicle = leg.rentedVehicle;
+                    }
+                    itin.legs.add(anonymizedLeg);
                 }
-                itin.legs.add(anonymizedLeg);
             }
             anonymizedItineraries.add(itin);
         }
         return anonymizedItineraries;
     }
 
     /**
-     * Create {@link Coordinates} for a leg. Replace lat/lon values with the lat/lon values created for the trip
-     * request if not a transit leg. The start and end legs will then be consistent with the trip's 'to' and 'from'
-     * place.
+     * Define the coordinates for all legs. If the leg is a walking leg and is before the first transit leg or after
+     * the last transit leg the coordinates must not be provided for privacy. Any transit leg or walking legs between
+     * transits legs, the coordinates can be provided.
      */
-    private Coordinates getLegCoordinates(
-        boolean isTransitLeg,
-        boolean isFirstOrLastLegOfTrip,
-        Place place,
-        Coordinates tripRequestCoordinates
-    ) {
-        return (!isTransitLeg && isFirstOrLastLegOfTrip) ? tripRequestCoordinates : new Coordinates(place.lat, place.lon);
+    private void processLegCoordinates(List<Leg> legs) {
+        int firstTransitLeg = getFirstTransitLeg(legs);
+        int lastTransitLeg = getLastTransitLeg(legs);
+        for (int i = 0; i <= legs.size() -1; i++) {
+            Leg leg = legs.get(i);
+            if (
+                leg.mode.equalsIgnoreCase("walk") &&
+                (i < firstTransitLeg || i > lastTransitLeg)
+            ) {
+                // The walking leg is before the first transit leg or after the last transit leg, remove the
+                // coordinates.
+                removeCoordinatesFromLeg(leg);
+            }
+        }
+    }
+
+    /**
+     * Define the position of the first transit leg. If all walking legs, return the number of legs plus one to
+     * represent all walk legs.
+     */
+    private int getFirstTransitLeg(List<Leg> legs) {
+        for (int i = 0; i <= legs.size(); i++) {
+            if (Boolean.TRUE.equals(legs.get(i).transitLeg)) {
+                return i;
+            }
+        }
+        return legs.size() + 1;
+    }
+
+    /**
+     * Define the position of the last transit leg. If all walking legs, return zero to represent all walk legs.
```

Comment:
The code will correctly also strip locations from a bike or escooter leg (since they are not transit legs) but the comment should reflect this as well